pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -326,6 +331,14 @@ func (s *NodeController) checkNodeReady(node *api.Node) *api.NodeCondition {
 	}
 }
 
+// sendConditionEvent sends condition change event.
+func (s *NodeController) sendConditionEvent(node *api.Node, kind api.NodeConditionKind, oldCondition, newCondition *api.NodeCondition) {
+	// TODO: if 'oldCondition' is nil, we should publish the initial condition, but selfLink is not set.
+	if oldCondition != nil {
+		record.Eventf(node, "condition change", "Node condition %v changes from %v to %v", kind, oldCondition.Status, newCondition.Status)
```

Comment:
No, my complaint is that "changedCondition" describes everything, so it describes nothing useful. To be useful, an event should be something like "started" or "failed" or "restarted"-- the reason should say something about the transition, not just indicate that a transition happened.